ipfs-api/src/client.rs

```diff
@@ -470,7 +470,38 @@ impl IpfsClient {
 
         form.add_reader("path", data);
 
-        self.request(&request::Add, Some(form))
+        self.request(&request::Add::default(), Some(form))
+    }
+
+    /// Add a file to IPFS with options.
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    /// # extern crate ipfs_api;
+    /// #
+    /// use ipfs_api::IpfsClient;
+    /// use std::io::Cursor;
+    ///
+    /// # fn main() {
+    /// let client = IpfsClient::default();
+    /// let data = Cursor::new("Hello World!");
+    /// let mut add = ipfs_api::request::Add::default();
+    /// add.chunker = Some("rabin-512-1024-2048");
+    /// let req = client.add_with_options(data, &add);
+    /// # }
+    /// ```
+    ///
+    #[inline]
+    pub fn add_with_options<R>(&self, data: R, add: &request::Add) -> AsyncResponse<response::AddResponse>
```

Comment:
Maybe make this take in `B: Borrow<request::Add>` instead to allow it to take an owned request?

Reply:
Ok, good idea.